### PR TITLE
AWS: move `self._wait_public_ip()` back to before `super()`

### PR DESCRIPTION
**Pull request by fruch**
Was causing `self.public_address` to return None
and failing the cluster setup like that, when trying to initalize
the ssh connection:

```
 Traceback (most recent call last):
   File "/sct/sdcm/tester.py", line 134, in wrapper
     return method(*args, **kwargs)
   File "/sct/sdcm/utils/common.py", line 136, in inner
     res = func(*args, **kwargs)
   File "/sct/sdcm/tester.py", line 258, in setUp
     self.init_resources()
   File "/sct/sdcm/tester.py", line 797, in init_resources
     monitor_info=monitor_info)
   File "/sct/sdcm/tester.py", line 618, in get_cluster_aws
     self.db_cluster = create_cluster(db_type)
   File "/sct/sdcm/tester.py", line 584, in create_cluster
     **cl_params)
   File "/sct/sdcm/cluster_aws.py", line 780, in __init__
     aws_extra_network_interface=params.get('aws_extra_network_interface'))
   File "/sct/sdcm/cluster.py", line 2575, in __init__
     super(BaseScyllaCluster, self).__init__(*args, **kwargs)
   File "/sct/sdcm/cluster_aws.py", line 95, in __init__
     node_type=node_type,
   File "/sct/sdcm/cluster.py", line 2373, in __init__
     self.add_nodes(num, dc_idx=dc_idx)
   File "/sct/sdcm/cluster_aws.py", line 803, in add_nodes
     enable_auto_bootstrap=enable_auto_bootstrap)
   File "/sct/sdcm/cluster_aws.py", line 385, in add_nodes
     enumerate(instances, start=self._node_index + 1)]
   File "/sct/sdcm/cluster_aws.py", line 384, in <listcomp>
     for node_index, instance in
   File "/sct/sdcm/cluster_aws.py", line 399, in _create_node
     base_logdir=base_logdir, dc_idx=dc_idx, node_type=self.node_type)
   File "/sct/sdcm/cluster_aws.py", line 429, in __init__
     dc_idx=dc_idx)
   File "/sct/sdcm/cluster.py", line 404, in __init__
     self.remoter = RemoteCmdRunner(**ssh_login_info)
   File "/sct/sdcm/remote.py", line 189, in __init__
     connect_kwargs=self.connect_config)
   File "/sct/sdcm/remote.py", line 106, in _create_connection
     self.connection = Connection(*args, **kwargs)
   File "/usr/local/lib/python3.6/site-packages/fabric/connection.py", line 386, in __init__
     shorthand = self.derive_shorthand(host)
   File "/usr/local/lib/python3.6/site-packages/fabric/connection.py", line 574, in derive_shorthand
     return derive_shorthand(host_string)
   File "/usr/local/lib/python3.6/site-packages/fabric/connection.py", line 34, in derive_shorthand
     user_hostport = host_string.rsplit("@", 1)
 AttributeError: 'NoneType' object has no attribute 'rsplit'
```

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I gave variables/functions meaningful self-explanatory names
- [ ] I didn't leave commented-out/debugging code
- [ ] I didn't copy-paste code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by fruch** [comment]
here @juliayakovlev, this should fix the issue you were having

**Comment by amoskong** [comment]
FYI,  @alexsavilov also moved the _wait_public_ip() like this PR.

https://github.com/scylladb/scylla-cluster-tests/pull/1488/files

File: sdcm/cluster_aws.py [hunk 1]
```diff
@@ -421,15 +421,20 @@ def __init__(self, ec2_instance, ec2_service, credentials, parent_cluster,  # py
                           'key_file': credentials.key_file}
         self._spot_aws_termination_task = None
 
+        if len(self._instance.network_interfaces) == 2:
```
Comment (amoskong) [inline sdcm/cluster_aws.py]: Is it correct?
do we want to only wait the public ip for instance that has two network interfaces?

Comment (fruch) [reply sdcm/cluster_aws.py]: It's incorrect, I wanted to reverse the condition

**Comment by bentsi** [comment]
The correct solution is a mix of this PR and @aleksbykov (not alexsavilov Amos!)
```
        self._wait_public_ip()
        super(AWSNode, self).__init__(name=name,
                                      parent_cluster=parent_cluster,
                                      ssh_login_info=ssh_login_info,
                                      base_logdir=base_logdir,
                                      node_prefix=node_prefix,
                                      dc_idx=dc_idx)
        if len(self._instance.network_interfaces) == 2:
            self.allocate_and_attach_elastic_ip()
            # when there are two network interfaces, public address isn't set automatically by AWS
            # it's a manual process we'll do later on
            self._wait_public_ip()
```

**Comment by bentsi** [comment]
@fruch put an urgent label on it

**Pull request merged**